src/assert.c

```diff
@@ -377,10 +381,20 @@ fido_get_signed_hash(int cose_alg, fido_blob_t *dgst,
 	authdata_len = cbor_bytestring_length(item);
 
 	if (cose_alg != COSE_EDDSA) {
-		if (dgst->len < SHA256_DIGEST_LENGTH || SHA256_Init(&ctx) == 0 ||
+		if (dgst->len < SHA256_DIGEST_LENGTH ||
+#if OPENSSL_VERSION_NUMBER < 0x30000000
+		    SHA256_Init(&ctx) == 0 ||
 		    SHA256_Update(&ctx, authdata_ptr, authdata_len) == 0 ||
 		    SHA256_Update(&ctx, clientdata->ptr, clientdata->len) == 0 ||
-		    SHA256_Final(dgst->ptr, &ctx) == 0) {
+		    SHA256_Final(dgst->ptr, &ctx) == 0
+#else
+		    (mdctx = EVP_MD_CTX_new()) == NULL ||
+		    EVP_DigestInit_ex(mdctx, EVP_sha256(), NULL) <= 0 ||
+		    EVP_DigestUpdate(mdctx, authdata_ptr, authdata_len) <= 0 ||
+		    EVP_DigestUpdate(mdctx, clientdata->ptr, clientdata->len) <= 0 ||
+		    EVP_DigestFinal_ex(mdctx, dgst->ptr, NULL) <= 0
```

Comment:
These functions are available in OpenSSL 3.0, 1.1.1, 1.1.0, and 1.0. Wouldn't it be better to switch to them entirely and drop the `#if`? IIRC, `EVP_MD_CTX` is initialised differently in 1.0. On that note, we could consider dropping OpenSSL 1.0 compatibility if it makes 3.0 adaptation easier as a whole.